app/components/Map.js

```diff
@@ -1,24 +1,94 @@
-import React, { Component, PropTypes, StyleSheet, Text, View  } from 'react-native';
+
+import React, { Component, PropTypes, StyleSheet, Text, View, Dimensions, AlertIOS  } from 'react-native';
+
+import MapView from 'react-native-maps';
+
 
 export default class Map extends Component {
   constructor(props) {
     super(props);
     this.state = {
+      position: {
+        latitude: 37.78825,
+        longitude: -122.4324,
+        latitudeDelta: 0.0922,
+        longitudeDelta: 0.0421,
+      },
+      lastPosition: undefined,
+      pins: [],
+      walker: undefined
     };
+
+    // this.currentPosition = new Firebase("https://interruptedlobster.firebaseio.com/testPosition");
   }
+
+  // listenForItems(itemsRef) {
+  //     itemsRef.on('value', (snap) => {
+  //       // get children as an array
+        
+  //       this.setState({
+  //         firebasePosition: snap
+  //       });
+  //     });
+  //   }
+
+  // componentDidMount() {
+  //   navigator.geolocation.getCurrentPosition(
+  //     (position) => {
+  //       var initialPosition = JSON.stringify(position);
+  //       this.setState({walker: initialPosition});
+  //     },
+  //     (error) => {
+  //       alert(error.message)
+  //     },
+  //     {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
+  //   );
+
+  //   this.watchID = navigator.geolocation.watchPosition(
+  //     (position) => {
+  //       var lastPosition = JSON.stringify(position);
+  //       // update firebase with current position
+  //       // this.currentPosition.set(position);
+  //       this.setState({lastPosition: position});
+  //     }
+  //   );
+  // }
+
+  onRegionChange(region) {
+    this.setState({ position: region });
+  }
+
   render() {
     return (
-     <View style={[styles.container, { backgroundColor: 'transparent'}]}>
-        <Text style={styles.welcome}>
-          Welcome to React Native!
-        </Text>
-        <Text style={styles.instructions}>
-          To get started, edit index.ios.js
-        </Text>
-        <Text style={styles.instructions}>
-          Press Cmd+R to reload,{'\n'}
-          Cmd+D or shake for dev menu
-        </Text>
+      <View style={styles.container}>
+        <MapView
+          showsUserLocation={true}
+          followUserLocation={true}
+          region={this.state.position}
+          onRegionChange={this.onRegionChange.bind(this)}
+          style={styles.map}
+          showsCompass={true}
+          onLongPress={
+            () =>
+            // this.currentPosition.push('hello');
+            // var event = JSON.stringify(e).split('\"').join('');
+            AlertIOS.alert(
+                'Drop a Pin?',
+                null,
+                [{
+                  text: 'Cancel',
+                  style: 'cancel'
+                },
+                {
+                  text: 'OK'
+                  // onPress: () => this.currentPosition.push(e)
+                }]
+              )
```

Comment:
Can this anonymous function be refactored out of the render() function? I think that would result in more modular code.